libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2278,27 +2316,33 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			auto const& toStructType = dynamic_cast<StructType const &>(_to);
 			solAssert(fromStructType.structDefinition() == toStructType.structDefinition(), "");
 
-			solUnimplementedAssert(!fromStructType.isDynamicallyEncoded(), "");
-			solUnimplementedAssert(toStructType.location() == DataLocation::Memory, "");
-			solUnimplementedAssert(fromStructType.location() != DataLocation::Memory, "");
-
-			if (fromStructType.location() == DataLocation::CallData)
-			{
-				body = Whiskers(R"(
-					converted := <abiDecode>(value, calldatasize())
-				)")("abiDecode", ABIFunctions(m_evmVersion, m_revertStrings, m_functionCollector).tupleDecoder(
-					{&toStructType}
-				)).render();
-			}
+			if (
+				fromStructType.location() == toStructType.location() &&
+				toStructType.location() != DataLocation::CallData &&
```

Comment:
If from location and to location is the same, why would calldata be a problem? Converting from calldata to calldata should work...

Reply:
That is true, will fix this.

Reply:
The problem will be multi-stack-slot calldata types.

Reply:
We could go ahead and have the function return ``<converted>`` and take ``<value>`` as arguments, allowing for multi-slot types - but maybe that's a bit much for the same PR here? We could also restrict to ``!dynamically sized`` in the calldata case for now.

Reply:
Then please add an assertion just about the number of slots

Reply:
I did it assertion `sizeOnStack() == 1`, hope it is fine like that